Source/LinqToDB.AspNet/ServiceConfigurationExtensions.cs

```diff
@@ -180,21 +181,32 @@ public static class ServiceConfigurationExtensions
 
 		private static bool HasTypedContextConstructor<TContext>()
 		{
-			var typedConstructorInfo   = typeof(TContext).GetConstructor(
-				BindingFlags.Public | BindingFlags.Instance | BindingFlags.ExactBinding,
-				null,
-				new[] {typeof(LinqToDBConnectionOptions<TContext>)},
-				null);
+			var constructors = typeof(TContext)
+				.GetConstructors(BindingFlags.Public | BindingFlags.Instance);
 
-			var untypedConstructorInfo = typedConstructorInfo == null
-				? typeof(TContext).GetConstructor(new[] {typeof(LinqToDBConnectionOptions) })
-				: null;
+			var typedConstructor = constructors
+				.Where(c => c.GetParameters()
+					.Where(p => p.ParameterType == typeof(LinqToDBConnectionOptions<TContext>))
+					.Count() == 1)
```

Comment:
```suggestion
					.Any(p => p.ParameterType == typeof(LinqToDBConnectionOptions<TContext>)))
```
I see no reason add the 'where count == 1', especially since there is no specific exception that describes the problem to the user.